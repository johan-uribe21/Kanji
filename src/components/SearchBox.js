import React from 'react';

class SearchBox extends React.Component  {
  constructor(props) {
    super(props);
    this.state = {
      input: ''
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  // updates the input field as the user types into input, and turns form into a controlled component.
  handleChange = (event) => {
    this.setState({input: event.target.value});
  };

  handleSubmit(event) {
    event.preventDefault();
    this.props.submitWord(this.state.input);
    this.setState({input: ''}); // resets the state.input value which resets the rendered value in input box
  };

  render() {
    return (
      <form 
        className = 'pa3 ba b--green bg-lightest-blue'
        onSubmit = {this.handleSubmit} >
        <label>Additional Vocabulary Words: 
          <input 
            id="inputKanji" 
            type="text" 
            size="20"
            onChange={this.handleChange}
            value = {this.state.input} // for controlled component, state controls values
            />
        </label>
        <input 
          type="submit" 
          value='submit'
          />
      </form>
    );
  }
}

export default SearchBox;