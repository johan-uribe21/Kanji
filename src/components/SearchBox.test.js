import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBox, {handleSubmit} from './SearchBox';
import React from 'react';

configure({ adapter: new Adapter() }); // need this line to use enzyme

describe('Component render, state updates onChange, and redux action onSubmit', () => {
  it('expect to render SearchBox component', () => {
    expect(shallow(<SearchBox />)).toMatchSnapshot();
  });

  it('updates the state when value is typed into search box', () => {
    const event = {
      preventDefault() {},
      target: {value: 'the-value'}
    }
    const wrapper = shallow(<SearchBox />);

    wrapper.find('#inputKanji').simulate('change', event);
    expect(wrapper.state()).toEqual({input: 'the-value'});
    //expect(onChangeMock).toBeCalledWith('the-value');
  });

  it('updates the state with onChange and then onSubmit passes state.input to redux action', () => {
    const updateReduxStoreAction = jest.fn();
    const event = {
      preventDefault() {},
      target: {value: 'the-value'}
    }
    const wrapper = shallow(<SearchBox submitWord={updateReduxStoreAction}/>);

    // simulates the onchange event with a synthetic event
    wrapper.find('#inputKanji').simulate('change', event);
    // simulates onSubmit of form after state has been updated with onChange.
    wrapper.find('#SearchBox').simulate('submit', event);
    // checks of onSubmit passes the correct parameters from state.input to submitWord function
    // The updateReduxStoreAction function is the redux action passed in the props to this component
    expect(updateReduxStoreAction).toBeCalledWith('the-value');
  });
});

