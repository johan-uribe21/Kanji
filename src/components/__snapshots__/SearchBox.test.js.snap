// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox render expect to render SearchBox component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Additional Vocabulary Words:
          <input
            id="inputKanji"
            onChange={[Function]}
            size="20"
            type="text"
            value=""
          />
        </label>,
        <input
          type="submit"
          value="submit"
        />,
      ],
      "className": "pa3 ba b--green bg-lightest-blue",
      "id": "SearchBox",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Additional Vocabulary Words:",
            <input
              id="inputKanji"
              onChange={[Function]}
              size="20"
              type="text"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Additional Vocabulary Words:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "inputKanji",
              "onChange": [Function],
              "size": "20",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
          "value": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Additional Vocabulary Words:
            <input
              id="inputKanji"
              onChange={[Function]}
              size="20"
              type="text"
              value=""
            />
          </label>,
          <input
            type="submit"
            value="submit"
          />,
        ],
        "className": "pa3 ba b--green bg-lightest-blue",
        "id": "SearchBox",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Additional Vocabulary Words:",
              <input
                id="inputKanji"
                onChange={[Function]}
                size="20"
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Additional Vocabulary Words:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "inputKanji",
                "onChange": [Function],
                "size": "20",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
            "value": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
